app/javascript/components/App.js

```diff
@@ -63,126 +59,181 @@ componentDidMount(){
-    },
-    method: "POST"
-    })
-    .then(response => response.json())
-    .then(payload => { 
-      //this line will return the crawl ID of the crawl
-      return payload.id
-      // payload.id 
-      //might be payload.crawl.id- depends what the console log below looks like
-    })
+createNewCrawl = (title, userid) => {
+  const requestBody = {
+    title: title,
+    user_id: userid
   }
+  return fetch('/crawls', {
+    body: JSON.stringify(requestBody),
+    headers: {
+    "Content-Type": "application/json"
+  },
+  method: "POST"
+  })
+  .then(response => response.json())
+  .then(payload => { 
+    //this line will return the crawl ID of the crawl
+    return payload.id
+    // payload.id 
+    //might be payload.crawl.id- depends what the console log below looks like
+  })
+}
 
 
-  addBartoCrawl = (bar, crawlID) => {
-    const requestBody = {
-      bar: {
-        yelp_id: bar.id, 
-        name: bar.name, 
-        image_url: bar.image_url, 
-        url: bar.url,
-        categories: bar.categories.map(category => category.title).join(" / "),
-        rating: bar.rating, 
-        address1: bar.location.address1, 
-        city: bar.location.city, 
-        state: bar.location.state,
-        zip_code: bar.location.zip_code, 
-        country: bar.location.country, 
-        display_address: bar.location.display_address.join(", "),
-        phone: bar.phone, 
-        display_phone: bar.display_phone
-      },
-      crawl: {
-        id: crawlID
-      }
-    }
-    return fetch('/bars', {
-      body: JSON.stringify(requestBody),
-      headers: {
-      "Content-Type": "application/json"
+addBartoCrawl = (bar, crawlID) => {
+  const requestBody = {
+    bar: {
+      yelp_id: bar.id, 
+      name: bar.name, 
+      image_url: bar.image_url, 
+      url: bar.url,
+      categories: bar.categories.map(category => category.title).join(" / "),
+      rating: bar.rating, 
+      address1: bar.location.address1, 
+      city: bar.location.city, 
+      state: bar.location.state,
+      zip_code: bar.location.zip_code, 
+      country: bar.location.country, 
+      display_address: bar.location.display_address.join(", "),
+      phone: bar.phone, 
+      display_phone: bar.display_phone
     },
-    // HTTP verb so the correct endpoint is invoked on the server
-      method: "POST"
-    })
-    .then(response => {
-      if(response.status === 422){
-        alert("Your submission was not accepted. BYE")
-      }
-      return response.json()
-    })
-    .catch(errors => {
-      console.log("Add Bar to Crawl Errors:", errors)
-    })
+    crawl: {
+      id: crawlID
+    }
   }
+  return fetch('/bars', {
+    body: JSON.stringify(requestBody),
+    headers: {
+    "Content-Type": "application/json"
+  },
+  // HTTP verb so the correct endpoint is invoked on the server
+    method: "POST"
+  })
+  .then(response => {
+    if(response.status === 422){
+      alert("Your submission was not accepted. BYE")
```

Comment:
lol this is gonna be a great user experience

Reply:
lol v important